x-pack/plugins/security_solution/public/management/cypress/e2e/artifacts.cy.ts

```diff
@@ -0,0 +1,183 @@
+        loginWithWriteAccess(`/app/security/administration/${testData.urlPath}`);
+        // Opens add flyout
+        cy.getBySel(`${testData.pagePrefix}-emptyState-addButton`).click();
+
+        for (const formAction of testData.create.formFields) {
+          if (formAction.type === 'click') {
+            cy.getBySel(formAction.selector).click();
+          } else if (formAction.type === 'input') {
+            cy.getBySel(formAction.selector).type(formAction.value || '');
+          } else if (formAction.type === 'customClick') {
+            cy.get(formAction.selector).click();
+          }
+        }
+
+        // Submit create artifact form
+        cy.getBySel(`${testData.pagePrefix}-flyout-submitButton`).click();
+
+        // Check new artifact is in the list
+        for (const checkResult of testData.create.checkResults) {
+          cy.getBySel(checkResult.selector).should('have.text', checkResult.value);
+        }
+
+        // Title is shown after adding an item
+        cy.getBySel('header-page-title').contains(testData.title);
+      });
+
+      it(`read - should not be able to update/delete an existing ${testData.title} entry`, () => {
+        loginWithReadAccess(`/app/security/administration/${testData.urlPath}`);
+        cy.getBySel('header-page-title').contains(testData.title);
+        cy.getBySel(`${testData.pagePrefix}-card-header-actions-button`).should('not.exist');
+        cy.getBySel(`${testData.pagePrefix}-card-cardEditAction`).should('not.exist');
+        cy.getBySel(`${testData.pagePrefix}-card-cardDeleteAction`).should('not.exist');
+      });
+
+      it(`read - should not be able to create a new ${testData.title} entry`, () => {
+        loginWithReadAccess(`/app/security/administration/${testData.urlPath}`);
+        cy.getBySel('header-page-title').contains(testData.title);
+        cy.getBySel(`${testData.pagePrefix}-pageAddButton`).should('not.exist');
+      });
+
+      it(`write - should be able to update an existing ${testData.title} entry`, () => {
+        loginWithWriteAccess(`/app/security/administration/${testData.urlPath}`);
+        // Opens edit flyout
+        cy.getBySel(`${testData.pagePrefix}-card-header-actions-button`).click();
+        cy.getBySel(`${testData.pagePrefix}-card-cardEditAction`).click();
+
+        for (const formAction of testData.update.formFields) {
+          if (formAction.type === 'click') {
+            cy.getBySel(formAction.selector).click();
+          } else if (formAction.type === 'input') {
+            cy.getBySel(formAction.selector).type(formAction.value || '');
+          } else if (formAction.type === 'clear') {
+            cy.getBySel(formAction.selector).clear();
+          } else if (formAction.type === 'customClick') {
+            cy.get(formAction.selector).click();
+          } else if (formAction.type === 'removeItem') {
+            cy.get(`[data-test-subj="${formAction.selector}"] > span > button`).click();
+          }
+        }
```

Comment:
This action performer is a superset of the other one in lines 104:112 - they could be extracted